Tests/SonarScanner.MSBuild.TFS.Tests/ProgramTests.cs

```diff
@@ -0,0 +1,123 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using TestUtilities;
+
+namespace SonarScanner.MSBuild.TFS.Classic.Tests
+{
+    [TestClass]
+    public class ProgramTests
+    {
+        [TestMethod]
+        public void Execute_WhenNoArgs_ShouldLogError()
+        {
+            var logger = new TestLogger();
+            var result = Program.Execute(new string[] { }, logger);
+
+            result.Should().Be(0);
+            logger.Warnings.Should().BeEmpty();
+            logger.Errors.Should().HaveCount(1);
+            logger.Errors.Should().Contain(new string[] { "No argument found. Exiting..." });
+        }
+
+        [TestMethod]
+        public void Execute_Method_Is_Uknown_Should_Log_Error()
+        {
+            var logger = new TestLogger();
+            var result = Program.Execute(new string[] { "MockMethod" }, logger);
+
+            result.Should().Be(0);
+            logger.Errors.Should().HaveCount(2);
+            logger.Errors.Should().Contain(new string[] { "An exception occured while executing the process : Requested value 'MockMethod' was not found." });
+        }
+
+        [TestMethod]
+        public void Execute_ShouldExecute_CoverageConverter_ShouldSucceeed()
+        {
+            var logger = new TestLogger();
+            var tempDir = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString())).FullName;
+            var text = "<?xml version=\"1.0\" encoding=\"utf-8\"?>" +
+                        "<AnalysisConfig xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\"" +
+                        " xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns=\"http://www.sonarsource.com/msbuild/integration/2015/1\">" +
+                        "</AnalysisConfig>";
+
+            File.WriteAllText(Path.Combine(tempDir, "temp.xml"), text);
+            File.WriteAllText(Path.Combine(tempDir, "sonar-project.properties"), string.Empty);
+
+            //Faking TeamBuild
+            Environment.SetEnvironmentVariable("TF_Build", "true");
+            Environment.SetEnvironmentVariable("BUILD_BUILDURI", "vsts://test/test");
+            Environment.SetEnvironmentVariable("AGENT_BUILDDIRECTORY", tempDir);
+            Environment.SetEnvironmentVariable("BUILD_SOURCESDIRECTORY", tempDir);
+
+            var result = Program.Execute(new string[] { "ConvertCoverage", Path.Combine(tempDir, "temp.xml"), Path.Combine(tempDir, "sonar-project.properties") }, logger);
+
+            result.Should().Be(0);
+            logger.Errors.Should().HaveCount(0);
+            logger.InfoMessages.Should().Contain("Coverage report conversion completed successfully.");
+            Environment.SetEnvironmentVariable("TF_Build", null);
```

Comment:
This cleanup won't work if the test fails. Fortunately, that's what the `EnvironmentVariableScope` test helper class is for. It's used by any tests that either need to temporarily set the environment variables to specific test values, or to make sure the test doesn't behave differently when run on the CI because of the environment variables set by Azdo (see e.g. `TryProcessArgsIsolatedFromEnvironment`).

Reply:
Thanks for that !